activation/nipost.go

```diff
@@ -159,7 +160,7 @@ func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.PoetC
 		if err != nil {
 			return nil, 0, err
 		}
-		signature := nb.signer.Sign(challenge)
+		signature := nb.signer.Sign(signing.ATX, challenge)
```

Comment:
@countvonzero this is what i was talking about. it is safer to use different domain here, otherwise we need to carefully consider what is the structure of this challenge and ensure that it can't be decoded as a valid atx metadata